src/fizz-buzz/fizz-buzz.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  buzzRule,
+  defaultRule,
+  fizzBuzzRule,
+  fizzRule,
+  Rule,
+  zeroRule,
+} from './rules';
+
+export class FizzBuzzTranslator {
+  rules: Rule[] = [];
+
+  constructor(rules?: Rule[]) {
+    if (rules) {
+      this.rules = rules;
+    } else {
+      this.rules.push(zeroRule);
+      this.rules.push(fizzBuzzRule);
+      this.rules.push(fizzRule);
+      this.rules.push(buzzRule);
+      this.rules.push(defaultRule);
+    }
+  }
+
+  translate(input: number): string {
+    let translation = '';
+
+    this.rules.forEach((rule) => (translation += rule(translation, input)));
+
+    return translation;
+  }
+
+  findContiguousFizzBuzzInRange(from: number, to: number): [number, number][] {
+    const contiguousFizzBuzz: [number, number][] = [];
+
+    [...Array(to - from).keys()]
```

Comment:
un test pour s'assurer que from < to permettrait de protéger l'appel de cette méthode